styleguide.md

```diff
@@ -0,0 +1,142 @@
+# Code Style Guides and Tips
+
+## Common Rules
+
+ - **maximum line length** should be < 80
+ - **indentation** should be 2 spaces
+ - **trim trailing whitespace** it would be great, if your IDE/text editor automatically trim trailing whitespaces
+
+## JS
+
+  - all variable declarations should have their **own variable keyword**, e.g.
+    ```javascript
+    // bad
+    const className = CLASSES['main-panel'],
+      isHiddenClassName = this._calcHiddenClassName(),
+      isResizingClassName = this.state.isResizing ? CLASSES['main-panel--is-resizing'] : '';
+
+
+    // good
+    const className = CLASSES['main-panel'];
+    const isHiddenClassName = this._calcHiddenClassName();
+    const isResizingClassName = this.state.isResizing ? CLASSES['main-panel--is-resizing'] : '';
+    ```
+
+- all **private methods** should be declared **with underscore** in front of the method name
+    ```javascript
+    // bad
+    toggleVisibility() {
+      ...
+    }
+
+
+    // good
+    _toggleVisibility() {
+      ...
+    }
+    ```
+
+## Preact
+
+- All **components** should be created as **ES6 classes** which extend from preact.Component
```

Comment:
Not very accurate - you can use pure functions instead of `es2015` classes:

```javascript
export default () => {
  return <div> </div>;
}
```
Will work just perfect if you don't need anything besides render props to markup.